src/game/engine/template.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Chronoshift is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#include "template.h"
+#include "cell.h"
+#include "globals.h"
+#include "iomap.h"
+#include "tileset.h"
+
+/**
+ * @brief
+ *
+ * 0x0056A1D8
+ */
+TemplateClass::TemplateClass(TemplateType type, cell_t cellnum) : ObjectClass(RTTI_TEMPLATE, g_Templates.ID(this))
+{
+    if (cellnum != -1) {
+        Unlimbo(Cell_To_Coord(cellnum));
+    }
+}
+
+/**
+ * @brief
+ *
+ * 0x0056A198
+ */
+void *TemplateClass::operator new(size_t size)
+{
+    TemplateClass *this_ptr = g_Templates.Alloc();
+    DEBUG_ASSERT(this_ptr != nullptr);
+    if (this_ptr != nullptr) {
+        this_ptr->m_IsActive = true;
+    }
+    return this_ptr;
+}
+
+/**
+ * @brief
+ *
+ * 0x0056A1B8
+ */
+void TemplateClass::operator delete(void *ptr)
+{
+    TemplateClass *this_ptr = static_cast<TemplateClass *>(ptr);
+    DEBUG_ASSERT(this_ptr != nullptr);
+    if (this_ptr != nullptr) {
+        this_ptr->m_IsActive = false;
+    }
+    g_Templates.Free(this_ptr);
+}
+
+/**
+ * @brief
+ *
+ * 0x004CDB90
+ */
+TemplateClass::~TemplateClass()
+{
+    if (GameActive) {
+        Limbo();
+    }
+}
+
+/**
+ * @brief
+ *
+ * 0x00569F24
+ */
+BOOL TemplateClass::Mark(MarkType mark)
+{
+    DEBUG_ASSERT(m_IsActive);
+    static bool _noup = false;
+
+    void *image = Class_Of().Get_Image_Data();
+    if (image == nullptr || ObjectClass::Mark(mark) == false) {
```

Comment:
Your formatting of the check against ObjectClass::Mark() here is overkill. Use "!ObjectClass::Mark()".